content/en/docs/concepts/overview/working-with-objects/names.md

```diff
@@ -18,14 +18,41 @@ For non-unique user-provided attributes, Kubernetes provides [labels](/docs/conc
 
 {{% /capture %}}
 
-
 {{% capture body %}}
 
 ## Names
 
 {{< glossary_definition term_id="name" length="all" >}}
 
-Kubernetes resources can have names up to 253 characters long. The characters allowed in names are: digits (0-9), lower case letters (a-z), `-`, and `.`.
+Below are three types of commonly used name constraints for resources.
+
+### DNS Subdomain Names
+
+Most resource types require a name that can be used as a DNS subdomain name
+as defined in [RFC 1123](https://tools.ietf.org/html/rfc1123).
+This means the name must:
+
+- contain no more than 253 characters
+- contain only lowercase alphanumeric characters, '-' or '.'
+- start with an alphanumeric character
+- end with an alphanumeric character
+
+### DNS Label Names
+
+There are some resource names are required to conform to the DNS
+label standard as defined in [RFC 1123](https://tools.ietf.org/html/rfc1123).
```

Comment:
@tengqm , changes look good. One nit, reword line 42. What do you think about these changes:
Some resources require their names to conform to the DNS label standard ...
Some resource types require their names to follow the DNS label standard ...

Reply:
Done